### ISSUE TITLE
Datagrids with clrDgSingleSelected and width constraints do not resize their columns correctly

### ISSUE THREAD

**Issue by ngenereux-ps** [vmware/clarity#2692]
## Describe the bug

When a datagrid uses the Single Select option (clrDgSingleSelected) and also has data which needs to have word-breaks applied to fit in the container size, the datagrid will be automatically sized to include a horizontal scrollbar. This happens even if the version of the datagrid without clrDgSingleSelected works correctly at a similar size.

## How to reproduce
[Stackblitz Example](https://stackblitz.com/edit/clarity-light-theme-v013-vj33sm)

1.  Create a datagrid within a div with a width constraint (say 500px)
2.  Add 5 columns with headers Column 1, 2, 3, 4, 5
3.  Add a single row with data which will require a line break. This can be data which requires a mid-word break, or data that simply results in long lines.
4.  Do not add any kind of Single Select options (i.e. do not include the clrDgSingleSelected attribute)
5. Load the page and observe that the datagrid is rendered correctly, and no horizontal scrollbar is introduced
6. Add the clrDgSingleSelected attribute to the datagrid
7. Increase the size of the div by 50px (this should be enough to accommodate the new column for the radio button, which as of 0.13.2 is 38px)
8. Reload the page and observe that even though the data columns in the grid have the same amount of space as before, a horizontal scrollbar is now shown.

## Expected behavior

If a datagrid can be displayed without a horizontal scrollbar when the clrDgSingleSelected option is not used, then that same datagrid should be displayed without a horizontal scrollbar when clrDgSingleSelected is used and it is given additional width to accommodate the radio buttons.

## Versions

**App**

* Angular: 6
* Node: 10.5.0
* Clarity: 0.13.2

**Device:**

* Type: MacBook
* OS: macOS High Sierra
* Browser: Chrome
* Version: 69

**Comment by ngenereux-ps** [comment]
Note: I tested https://github.com/vmware/clarity/pull/2670 with my sample project locally, and it does not address this issue.

**Comment by hippee-lee** [comment]
I'm closing because this feature is behaving as expected. Please read on below for why its behaving as expected and let me know if I have misunderstood your use case. 

In your working example the **five** columns have the following auto calculated widths: 

- Column 1: 113px
- Column 2: 97px
- Column 3: 96px
- Column 4: 96px
- Column 5: 96px

Collectively this adds up to 498 which, with the two 1px side borders = 500px and it fits nicely into the 500px wide container. 

The other two examples calculate exactly the same widths for the content projected. However, because **the single selection feature is declared it adds an additional column** of width ~38px and this additional column width added to the overall datagrid width causes the content to overflow and the horizontal scrollbar is shown. 

To summarize, 

1. In your `working` example you have five columns that add up to 500px width for the datagrid and it fits in the 500px wide container. 
2. In the `broken` example you have six columns that add up to ~538px width for the datagrid and it overflows the 500px container.

**Comment by ngenereux-ps** [comment]
I don't think that is the correct explanation. With that logic, I should be able to set the width of the div to say 550px, and have everything work. However, if you look at my updated example, you will see that even with this change the same problem exists. The bug is that with Single Selection and word wrapping enabled, the automatic sizer will size the columns within a few pixels of the container size (in the first example, 500px), and then add the 38 pixel selection column which creates the scrollbar.

**Comment by ngenereux-ps** [comment]
In fact, doing a little more testing, the scrollbar appears all the way up to 800px width on the `broken` datagrids. Given that without the 38px column, the `working` grid is properly sized at 500px, I think this is indeed a bug. Please let me know if you'd like to re-open this issue, or if you'd prefer that I log another issue with a more complete description.

**Comment by ngenereux-ps** [comment]
Thanks! I've tried to make the writeup more descriptive. It's still the same stackblitz, but I changed the width of the `broken` example to 550px, and changed the link to go to the edit view so it's easier to play around with and understand the issue.

**Comment by hippee-lee** [comment]
I looked into this a little more deeply and you are correct, when the column widths and cells are calculated the row actions are hidden and this causes the error. I'll try to get a fix in soon. Please subscribe to this issue for updates on the fix.
